### PR TITLE
:hatching_chick:  Use new spring authorization server 

### PR DESCRIPTION
**Pull request by blabadi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,16 +34,6 @@
...
-        <dependency>
-            <groupId>org.springframework.security.oauth</groupId>
-            <artifactId>spring-security-oauth2</artifactId>
-            <version>2.5.0.RELEASE</version>
-        </dependency>
-        <dependency>
-            <groupId>org.springframework.security.oauth.boot</groupId>
-            <artifactId>spring-security-oauth2-autoconfigure</artifactId>
-            <version>2.4.4</version>
-        </dependency>
```
Comment (blabadi) [inline pom.xml:46]: out with the old

File: pom.xml [hunk 2]
```diff
@@ -60,6 +50,12 @@
             <artifactId>spring-boot-starter-oauth2-client</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-authorization-server</artifactId>
+            <version>0.2.1</version>
+        </dependency>
+
```
Comment (blabadi) [inline pom.xml:58]: in with the new

File: src/main/java/bio/overture/ego/service/ApplicationService.java [hunk 3]
```diff
@@ -202,20 +200,43 @@ public Application getWithRelationships(@NonNull UUID id) {
...
+
+  @Override
+  public RegisteredClient findById(String s) {
+    return null;
+  }
+
+  @Override
+  public RegisteredClient findByClientId(String clientId) {
+    val application = getByClientId(clientId);
+    if (application == null) {
```
Comment (anncatton) [inline src/main/java/bio/overture/ego/service/ApplicationService.java:214]: `isNull()`

**Review by anncatton — APPROVED** [review]
Just need to fix first user test! 🚀

**Pull request merged**